research/src/fetch_campaign_donations.py

```diff
@@ -0,0 +1,346 @@
+import pandas as pd
+import urllib.request
+import os        
+import sys
+import zipfile
+import time
+
+def reporthook(blocknum, blocksize, totalsize):
+    readsofar = blocknum * blocksize
+    if totalsize > 0:
+        percent = readsofar * 1e2 / totalsize
+        s = "\r%5.1f%% %*d / %d" % (
+            percent, len(str(totalsize)), readsofar, totalsize)
+        sys.stderr.write(s)
+        if readsofar >= totalsize: # near the end
+            sys.stderr.write("\n")
+    else: # total size is unknown
+        sys.stderr.write("read %d\n" % (readsofar,))
+        
+url = 'http://agencia.tse.jus.br/estatistica/sead/odsele/prestacao_contas/prestacao_contas_2010.zip'
+file_name = 'prestacao_contas_2010.zip'
+print("Downloading 2010 donation data...")
+urllib.request.urlretrieve(url, file_name, reporthook)
+
+print("Uncompressing downloaded data.")
+with zipfile.ZipFile('prestacao_contas_2010.zip',"r") as zip_ref:
+    zip_ref.extractall("prestacao_contas_2010")
+os.remove("prestacao_contas_2010.zip")
+
+url = 'http://agencia.tse.jus.br/estatistica/sead/odsele/prestacao_contas/prestacao_final_2012.zip'
+file_name = 'prestacao_final_2012.zip'
+print("Downloading 2012 donation data...")
+urllib.request.urlretrieve(url, file_name, reporthook)
+
+print("Uncompressing downloaded data.")
+with zipfile.ZipFile('prestacao_final_2012.zip',"r") as zip_ref:
+    zip_ref.extractall("prestacao_final_2012")
+os.remove("prestacao_final_2012.zip")
+
+url = 'http://agencia.tse.jus.br/estatistica/sead/odsele/prestacao_contas/prestacao_final_2014.zip'
+file_name = 'prestacao_final_2014.zip'
+print("Downloading 2014 donation data...")
+urllib.request.urlretrieve(url, file_name, reporthook)
+
+print("Uncompressing downloaded data.")
+with zipfile.ZipFile('prestacao_final_2014.zip',"r") as zip_ref:
+    zip_ref.extractall("prestacao_final_2014")
+os.remove("prestacao_final_2014.zip")
+
+url = 'http://agencia.tse.jus.br/estatistica/sead/odsele/prestacao_contas/prestacao_contas_final_2016.zip'
+file_name = 'prestacao_contas_final_2016.zip'
+print("Downloading 2016 donation data...")
+urllib.request.urlretrieve(url, file_name, reporthook)
+
+print("Uncompressing downloaded data.")
+with zipfile.ZipFile('prestacao_contas_final_2016.zip',"r") as zip_ref:
+    zip_ref.extractall("prestacao_contas_final_2016")
+os.remove("prestacao_contas_final_2016.zip")
+
+print("Reading all the data and creating the dataframes...")
+
+donations_data_candidates_2010 = []
+donations_data_parties_2010 = []
+donations_data_committees_2010 = []
+for root, dirs, files in os.walk("prestacao_contas_2010/", topdown=False):
```

Comment:
It's not recommended to use slashes (`/`) in file paths because it changes from OS to OS (Windows, for example, uses `\` instead of `/`). Opt for `os.path.join` or if you really need the slash `os.sep`.

Reply:
BTW you shoud review the whole script to address this issue in different parts of it, ok?